osu.Game.Rulesets.Osu.Tests/TestSceneSliderInput.cs

```diff
@@ -26,14 +25,53 @@ namespace osu.Game.Rulesets.Osu.Tests
     public partial class TestSceneSliderInput : RateAdjustedBeatmapTestScene
     {
         private const double time_before_slider = 250;
-        private const double time_slider_start = 1500;
+        private const double time_slider_start = 0;
         private const double time_during_slide_1 = 2500;
         private const double time_during_slide_2 = 3000;
         private const double time_during_slide_3 = 3500;
         private const double time_during_slide_4 = 3800;
         private const double time_slider_end = 4000;
 
-        private List<JudgementResult> judgementResults;
+        private ScoreAccessibleReplayPlayer currentPlayer = null!;
+
+        private const float slider_path_length = 25;
+
+        private readonly List<JudgementResult> judgementResults = new List<JudgementResult>();
+
+        [TestCase(300, false)]
+        [TestCase(200, true)]
+        [TestCase(150, true)]
+        [TestCase(120, true)]
+        [TestCase(60, true)]
+        [TestCase(10, true)]
```

Comment:
Would probably add a few more test cases just for fuller coverage:

```diff
diff --git a/osu.Game.Rulesets.Osu.Tests/TestSceneSliderInput.cs b/osu.Game.Rulesets.Osu.Tests/TestSceneSliderInput.cs
index 5ee72757d5..498399fd84 100644
--- a/osu.Game.Rulesets.Osu.Tests/TestSceneSliderInput.cs
+++ b/osu.Game.Rulesets.Osu.Tests/TestSceneSliderInput.cs
@@ -44,6 +44,8 @@ public partial class TestSceneSliderInput : RateAdjustedBeatmapTestScene
         [TestCase(120, true)]
         [TestCase(60, true)]
         [TestCase(10, true)]
+        [TestCase(0, true)]
+        [TestCase(-30, false)]
         public void TestTailLeniency(float finalPosition, bool hit)
         {
             performTest(new List<ReplayFrame>

```

Reply:
Added one, added the other commented out.